File: BCC221/semana06-tp01/Casa.cpp
#include "Casa.h"

Casa :: Casa(const int id, const float valor, const string proprietario,
                        const string rua, const string bairro, const string cidade, const int numero,
                        const int quartos, const int banheiros, const int andares, const bool sala_jantar):
                        Imovel(id, valor, proprietario, rua, bairro, cidade, numero, quartos, banheiros),
                        andares(andares), sala_jantar(sala_jantar){}

int Casa::getAndares() const{
    return andares;
}

void Casa::setAndares(int andares){
    this->andares = andares;
}

bool Casa::getSalaJantar() const{
    return sala_jantar;
}

void Casa::setSalaJantar(bool sala_jantar){
    this->sala_jantar = sala_jantar;
}

void Casa::saida(ostream& out){
    Imovel::saida(out);
    out << "    Andares: " << this->getAndares() << endl;
}

ostream& operator<<(ostream& out, Casa& obj){
    obj.saida(out);
	return out;
}


void Casa::imprimirTudo(){
    cout << "    Sala de jantar: " << (this->getSalaJantar() ? "SIM" : "NAO") << endl;
}

Casa::~Casa() = default;

File: BCC221/semana14-tp02/src/main/java/backend/Casa.java
package backend;

public class Casa extends Imovel {

    private int andares;
    private boolean sala_jantar;

     //Construtor geral
    public Casa(double valor, String proprietario, String rua, String bairro, String cidade, int numero, int quartos, int banheiros, int andares, boolean sala_jantar) {
        super(valor, proprietario, rua, bairro, cidade, numero, quartos, banheiros);
        this.andares = andares;
        this.sala_jantar = sala_jantar;
    }

    //Construtor utilizando imovel e demais atributos
    public Casa(Imovel imovel, int andares, boolean sala_jantar) {
        super(imovel);
        this.andares = andares;
        this.sala_jantar = sala_jantar;
    }

    //Construtor inicializando padrões 
    public Casa() {
        super(0.0, "", "", "", "", 0, 0, 0);
        this.andares = 0;
        this.sala_jantar = false;
    }

    public int getAndares() {
        return andares;
    }

    public void setAndares(int andares) {
        if (andares > 0) {
            this.andares = andares;
        }
    }

    public boolean isSala_jantar() {
        return sala_jantar;
    }

    public void setSala_jantar(boolean sala_jantar) {
        this.sala_jantar = sala_jantar;
    }

    //Sobreescrita do método toString
    @Override
    public String toString() {
        return super.toString() + "Andares: " + this.andares;
    }

    //Sobreescrita do método que imprime atributos específicos da classe
    @Override
    public String imprimirClasse() {
        return super.toString() + "\n\tCaracterística Especifica\n"
                + "\t\tAndares: " + this.andares
                + "\n\t\tTem sala de jantar: " + (this.sala_jantar ? "SIM" : "NAO");
    }
}
